### PR TITLE
Gender bias filter integration

### PR DESCRIPTION
**Pull request by asnota**

### CODE REVIEW

**Review by AbinayaM02 — APPROVED** [review]
This is a duplicate of the #173 which is already approved.

File: filters/gender_bias/filter.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        neutral_group = [
+            flag.get("sentence")
+            for flag in flagged_corpus
+            if flag.get("neutral_flag") is True
+        ]
+
+        return feminine_group, masculine_group, neutral_group
+
+    def filter(self, sentences: []) -> bool:
+        biased = False
```
Comment (aadesh11) [inline filters/gender_bias/filter.py:326]: We don't need this as we are already setting the value below. (in line 344-346)

File: filters/gender_bias/filter.py [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+        f.close()
+
+        assert (
+            len(sentences) > 0
+        ), "You must provide at least one sentence for the analysis. Check the content of your sentences array you pass to the filter() method."
+
+        for sentence in sentences:
+
+            # Initialize the variables
+            feminine_flag = False
```
Comment (aadesh11) [inline filters/gender_bias/filter.py:188]: Line 188-193: Can we remove this initialization as we are already setting the same in the below code.

File: filters/gender_bias/filter.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+        self.language = language
+        self.feminine_input = feminine_input
+        self.masculine_input = masculine_input
+
+    @staticmethod
+    def flag_sentences(
+        sentences, language, feminine_input=[], masculine_input=[]
+    ):
+        """
+        flag sentences as belonging to the feminine, masculine or neutral groups
```
Comment (aadesh11) [inline filters/gender_bias/filter.py:22]: Please add docstrings for 'feminine_input' and 'masculine_input' param.
Also, add docstrings for class.

File: filters/gender_bias/filter.py [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+        :param sentences: sentences array
+        :param language: the string key representing the supported langage
```
Comment (aadesh11) [inline filters/gender_bias/filter.py:24]: Minor Cosmetic Change: 'langage' -> 'language'

File: filters/gender_bias/test.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
+{
```
Comment (aadesh11) [inline filters/gender_bias/test.json:1]: Please format the JSON file as indentation is not proper.

File: filters/gender_bias/filter.py (hunk 1 above)
Comment (asnota) [reply filters/gender_bias/filter.py]: Fixed.

File: filters/gender_bias/filter.py (hunk 2 above)
Comment (asnota) [reply filters/gender_bias/filter.py]: Initialization has been removed.

File: filters/gender_bias/filter.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+    @staticmethod
+    def count_genders(flagged_corpus):
+        """
+        count the number of sentences in each of groups
+        :param flagged_corpus: array of flagged sentences
+        :return: 3 integer values, representing feminine, masculine and neutral groups respectively
+        """
+        feminine_count = len(
+            [
+                flag
```
[the remark this answers is not in the document]
Comment (asnota) [reply filters/gender_bias/filter.py]: Added two generic methods to eliminate the code duplication.

File: filters/gender_bias/test.json (hunk 5 above)
Comment (asnota) [reply filters/gender_bias/test.json:1]: Fixed the identation.

Comment (asnota) [reply filters/gender_bias/test.json:1]: Hi @aadesh11, thank you for reading the code and providing valuable feedback, which resulted in a code improvement!

**Review by aadesh11 — APPROVED** [review]

**Review by aadesh11 — APPROVED** [review]

**Pull request merged**